File: frontend/src/api/route.ts
import type { OptimalPath, UID } from "./types";
import axios from "./index";

export function routeGetOrder(chainUID: UID) {
  return axios.get<UID[]>("/v2/route/order", {
    params: { chain_uid: chainUID },
  });
}

export function routeSetOrder(chainUID: UID, userUIDs: UID[]) {
  return axios.post<never>("/v2/route/order", {
    chain_uid: chainUID,
    route_order: userUIDs,
  });
}

export function routeOptimizeOrder(chainUID: UID) {
  return axios.get<OptimalPath>("/v2/route/optimize", {
    params: { chain_uid: chainUID },
  });
}

export interface RouteCoordinate {
  user_uid: string;
  latitude: number;
  longitude: number;
  route_order: number;
}

export function routeCoordinates(chainUID: UID) {
  return axios.get<RouteCoordinate[]>("/v2/route/coordinates", {
    params: { chain_uid: chainUID },
  });
}

File: server/sharedtypes/route.go
package sharedtypes

type RouteOrderSet struct {
	ChainUID   string   `json:"chain_uid" binding:"required"`
	RouteOrder []string `json:"route_order" binding:"required"`
}

type RouteCoordinatesGetResponseItem struct {
	UserUID    string  `json:"user_uid"`
	Latitude   float64 `json:"latitude"`
	Longitude  float64 `json:"longitude"`
	RouteOrder int     `json:"route_order"`
}
